runtime/runtime.go

```diff
@@ -123,6 +164,40 @@ func (r CommonRuntime) BuildBootCmd(bootCmd string) (string, error) {
 	return newBootCmd, nil
 }
 
+// inheritBootCmd builds boot command based on the package referenced by "base".
+func (r CommonRuntime) inheritBootCmd(cmdConfs map[string]*CmdConfig) (string, error) {
+	parts := strings.SplitN(r.Base, ":", 2)
+	pkgName := parts[0]
+	configSet := parts[1]
+
+	if _, exists := cmdConfs[pkgName]; !exists || cmdConfs[pkgName] == nil {
+		return "", fmt.Errorf("Failed to inherit from '%s': package not included or has no meta/run.yaml", pkgName)
+	}
+	if _, exists := cmdConfs[pkgName].ConfigSets[configSet]; !exists {
+		return "", fmt.Errorf("Failed to inherit '%s:%s': config_set does not exist", pkgName, configSet)
+	}
+
+	original := cmdConfs[pkgName].ConfigSets[configSet]
+
+	// Force my environment variables into inherited runtime.
+	env_orig := util.DeepCopyMap(original.GetEnv())
+	updatedKeys := original.ForceEnv(r.Env)
+	bootCmd, err := original.GetBootCmd(cmdConfs)
```

Comment:
I wonder if all this `deep copy-force-getboot-force` could be replaced with an update to `GetBootCmd` which would get the second parameter and would simply create the command either from the original env or from this env?